en/guide/typescript.md

```diff
@@ -131,6 +131,132 @@ export default class FeedPage extends Vue {
 
 You can use exact same logic for `layouts`.
 
+### Server Configuration with Express
+
+To use TypeScript in your server file you will need to update the following:
+
+> - Rename server/index.js to server/index.ts
+> - Replace all `require` statements with `import` statements inside the server/index.ts file
+> - Remove the ".js" file extension from the `nuxt.config.js` import 
+
+Here is an example
+
+```ts
+/* server/index.ts */
+import express from 'express'
+import consola from 'consola'
+import { Nuxt, Builder } from 'nuxt'
+const app = express()
+
+// Import and Set Nuxt.js options
+import config from '../nuxt.config'
+config.dev = !(process.env.NODE_ENV === 'production')
+
+async function start() {
+  // Init Nuxt.js
+  const nuxt = new Nuxt(config)
+
+  const { host, port } = nuxt.options.server
+
+  // Build only in dev mode
+  if (config.dev) {
+    const builder = new Builder(nuxt)
+    await builder.build()
```

Comment:
Why are you awaiting this step, in your medium article you didn't - and as you mentioned in your article: the server will be waiting for this and there would be no loading bars in your browser. I've also tested this behavior.

Reply:
Even for dev mode, there are many async hooks which if any failing we need to properly catch exceptions and also ensure code is loaded. Promise will be resolved as soon as watcher starts so this usage is OK. https://github.com/nuxt/nuxt.js/blob/dev/packages/webpack/src/builder.js#L139